tests/Happyr/LinkedIn/LinkedInTest.php

```diff
@@ -35,6 +35,22 @@ public function testConstructor()
             'Expect the API secret to be set.');
     }
 
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessageRegExp #.*\$storage.*DataStorageInterface.*#
+     */
+    public function testConstructorThrowsExceptionIfWrongStorageTypeIsPassed()
+    {
+        new LinkedIn(self::APP_ID, self::APP_SECRET, new \stdClass());
+    }
+    
+    public function testInitProperlyAssignsPassedInStorage()
```

Comment:
Great test! Do also test that the default `SessionStorage` is used when we don't have a third parameter.